extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxCoreRecorder.java

```diff
@@ -253,9 +204,34 @@ public static Vertx initialize(VertxConfiguration conf, VertxOptionsCustomizer c
 
         Vertx vertx;
 
+        Optional<ClassLoader> nonDevModeTccl;
```

Comment:
This whole block looks very similar to `createThreadFactory()`, no?

IMO, the complexity warrants a single method that can be used in both cases, but YMMV!